dss/operations/sync.py

```diff
@@ -0,0 +1,110 @@
+"""
+Replication consistency checks: verify and repair synchronization across replicas
+"""
+import json
+import typing
+import logging
+import argparse
+
+from collections import namedtuple
+from cloud_blobstore import BlobNotFoundError
+
+from dss import Config, Replica
+from dss.storage.hcablobstore import compose_blob_key
+from dss.operations import dispatch
+from dss.storage.identifiers import BLOB_PREFIX, FILE_PREFIX, BUNDLE_PREFIX
+
+
+logger = logging.getLogger(__name__)
+sync = dispatch.target("sync", help=__doc__)
+ReplicationAnomaly = namedtuple("ReplicationAnomaly", "key anomaly")
+
+
+def _log_warning(**kwargs):
+    logger.warning(json.dumps(**kwargs))
+
+
+@sync.action("verify-entity-replication",
+             arguments={"--source-replica": dict(choices=[r.name for r in Replica], required=True),
+                        "--destination-replica": dict(choices=[r.name for r in Replica], required=True),
+                        "--keys": dict(default=None, nargs="*", help="keys to check.")})
+def verify_entity_replication(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Verify replication for a DSS entity, following references.
+    for example, if a bundle key is provided, replication will be verified for the bundle and all referenced files and
+    blobs.
+    """
+    src_replica = Replica[args.source_replica]
```

Comment:
Could there be a check here such that, if `args.source_replica == args.destination_replica` it warns the user and stops before going through the entire operation of checking itself against itself?

Reply:
fixed